src/renderer/components/dock/pod-logs.store.ts

```diff
@@ -41,47 +46,106 @@ export class PodLogsStore extends DockTabStore<IPodLogsData> {
-      this.logs.set(tabId, { oldLogs: "", newLogs: "" })
-    }
-    const data = this.getData(tabId);
-    const { oldLogs, newLogs } = this.logs.get(tabId);
-    const { selectedContainer, tailLines, previous } = data;
-    const pod = new Pod(data.pod);
     try {
-      // if logs already loaded, check the latest timestamp for getting updates only from this point
-      const logsTimestamps = this.getTimestamps(newLogs || oldLogs);
-      let lastLogDate = new Date(0);
-      if (logsTimestamps) {
-        lastLogDate = new Date(logsTimestamps.slice(-1)[0]);
-        lastLogDate.setSeconds(lastLogDate.getSeconds() + 1); // avoid duplicates from last second
-      }
-      const namespace = pod.getNs();
-      const name = pod.getName();
-      const loadedLogs = await podsApi.getLogs({ namespace, name }, {
-        sinceTime: lastLogDate.toISOString(),
-        timestamps: true,  // Always setting timestampt to separate old logs from new ones
-        container: selectedContainer.name,
-        tailLines,
-        previous
+      const logs = await this.loadLogs(tabId, {
+        tailLines: this.lines + logRange
       });
-      if (!oldLogs) {
-        this.logs.set(tabId, { oldLogs: loadedLogs, newLogs });
-      } else {
-        this.logs.set(tabId, { oldLogs, newLogs: loadedLogs });
-      }
+      if (!this.refresher.isRunning) this.refresher.start();
+      this.logs.set(tabId, logs);
     } catch ({error}) {
-      this.logs.set(tabId, {
-        oldLogs: [
-          _i18n._(t`Failed to load logs: ${error.message}`),
-          _i18n._(t`Reason: ${error.reason} (${error.code})`)
-        ].join("\n"),
-        newLogs
-      });
+      const message = [
+        _i18n._(t`Failed to load logs: ${error.message}`),
+        _i18n._(t`Reason: ${error.reason} (${error.code})`)
+      ].join("\n");
+      this.refresher.stop();
+      this.logs.set(tabId, message);
+    }
+  }
+
+  /**
+   * Function is used to refreser/stream-like requests.
+   * It changes 'sinceTime' param each time allowing to fetch logs
+   * starting from last line recieved.
+   * @param tabId
+   */
+  loadMore = async (tabId: TabId) => {
+    const oldLogs = this.logs.get(tabId);
+    const logs = await this.loadLogs(tabId, {
+      sinceTime: this.getLastSinceTime(tabId)
+    });
+    // Add newly received logs to bottom
+    this.logs.set(tabId, oldLogs + logs);
+  }
+
+  /**
+   * Main logs loading function adds necessary data to payload and makes
+   * an API request
+   * @param tabId
+   * @param params request parameters described in IPodLogsQuery interface
+   * @returns {Promise} A fetch request promise
+   */
+  loadLogs = async (tabId: TabId, params: Partial<IPodLogsQuery>) => {
+    const data = this.getData(tabId);
+    const { selectedContainer, previous } = data;
+    const pod = new Pod(data.pod);
+    const namespace = pod.getNs();
+    const name = pod.getName();
+    return await podsApi.getLogs({ namespace, name }, {
```

Comment:
Unless you are trying to catch an error here, this should be `return podsApi.getLogs({...` so as to avoid the wrap/unwrap of the promise.